### PR TITLE
show not supported message for backup for Azure

### PR DESCRIPTION
**Pull request by udeeshagautam**
Fixes : #5628 

This PR contains 
1. Show _not supported_ messages for backup and restore if opened from command pallet for an azure db
2. Adding profile creation if not present for restore (same as backup) for robustness
3. reordering some code to reduce unnecessary calls.

### CODE REVIEW

File: src/sql/workbench/common/actions.ts [hunk 1]
```diff
@@ -239,16 +239,23 @@ export class BackupAction extends Task {
...
 			profile = TaskUtilities.getCurrentGlobalConnection(objectExplorerService, connectionManagementService, workbenchEditorService);
 		}
-		let configurationService = accessor.get<IConfigurationService>(IConfigurationService);
-		let previewFeaturesEnabled: boolean = configurationService.getValue('workbench')['enablePreviewFeatures'];
-		if (!previewFeaturesEnabled) {
-			return accessor.get<INotificationService>(INotificationService).info(nls.localize('backup.isPreviewFeature', 'You must enable preview features in order to use backup'));
+		if (profile) {
+			let serverInfo = connectionManagementService.getServerInfo(profile.id);
+			if (serverInfo && serverInfo.isCloud) {
+				return accessor.get<INotificationService>(INotificationService).info(nls.localize('backup.commandNotSupported', 'Backup command is not supported for Azure Sql databases.'));
```
Comment (kburtram) [inline src/sql/workbench/common/actions.ts]: should 'Sql' be spelled `SQL`?

Comment (udeeshagautam) [reply src/sql/workbench/common/actions.ts]: Good point.. also should it be SQL Azure or Azure SQL :)? The former right?

File: src/sql/workbench/common/actions.ts [hunk 2]
```diff
@@ -239,16 +239,23 @@ export class BackupAction extends Task {
 	}
 
 	runTask(accessor: ServicesAccessor, profile: IConnectionProfile): void | Promise<void> {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/common/actions.ts]: Unrelated to your PR...but is returning void | Promise<void> actually something people do? That seems really messy and makes handling the return type difficult - why wouldn't we just wrap a void return in a Promise.resolve instead?

File: src/sql/workbench/common/actions.ts [hunk 3]
```diff
@@ -239,16 +239,23 @@ export class BackupAction extends Task {
...
+		let configurationService = accessor.get<IConfigurationService>(IConfigurationService);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/common/actions.ts]: nit: But you should try to use const where you can. These should all be able to be const.

Comment (udeeshagautam) [reply src/sql/workbench/common/actions.ts]: not new code, just moved :) will change

File: src/sql/workbench/common/actions.ts (hunk 1 above)
Comment (udeeshagautam) [reply src/sql/workbench/common/actions.ts]: ok no, its **Azure SQL** at most places... so will keep that order..
image

**Pull request merged**